### ISSUE TITLE
make error in v2.1.2: Subindex

### ISSUE THREAD

**Issue by smfarzaneh** [dafer45/TBTK#8]
The following error occurs while trying to install v2.1.2

Sys info: 
- MacOS 10.14.6
- TBTK version v2.1.2
- cmake version 3.17.0
- GNU Make 3.81 

cmake seems to be working fine. But make does not. 
Here is the error after make command: 

```shell
Scanning dependencies of target TBTK
[  1%] Building CXX object Lib/src/CMakeFiles/TBTK.dir/Builders/ModelFactory.cpp.o
~/TBTK/Lib/src/Builders/ModelFactory.cpp:394:30: error: 
      non-constant-expression cannot be narrowed from type 'int' to
      'std::__1::vector<TBTK::Subindex, std::__1::allocator<TBTK::Subindex>
      >::size_type' (aka 'unsigned long') in initializer list
      [-Wc++11-narrowing]
                        vector<Subindex> newFrom({(int)n});
                                                  ^~~~~~
~/TBTK/Lib/src/Builders/ModelFactory.cpp:394:30: note: insert an
      explicit cast to silence this issue
                        vector<Subindex> newFrom({(int)n});
                                                  ^~~~~~
                                                  static_cast<size_type>( )
~/TBTK/Lib/src/Builders/ModelFactory.cpp:394:29: warning: braces
      around scalar initializer [-Wbraced-scalar-init]
                        vector<Subindex> newFrom({(int)n});
                                                 ^~~~~~~~
~/TBTK/Lib/src/Builders/ModelFactory.cpp:398:28: error: 
      non-constant-expression cannot be narrowed from type 'int' to
      'std::__1::vector<TBTK::Subindex, std::__1::allocator<TBTK::Subindex>
      >::size_type' (aka 'unsigned long') in initializer list
      [-Wc++11-narrowing]
                        vector<Subindex> newTo({(int)n});
                                                ^~~~~~
~/TBTK/Lib/src/Builders/ModelFactory.cpp:398:28: note: insert an
      explicit cast to silence this issue
                        vector<Subindex> newTo({(int)n});
                                                ^~~~~~
                                                static_cast<size_type>( )
~/TBTK/Lib/src/Builders/ModelFactory.cpp:398:27: warning: braces
      around scalar initializer [-Wbraced-scalar-init]
                        vector<Subindex> newTo({(int)n});
                                               ^~~~~~~~
~/TBTK/Lib/src/Builders/ModelFactory.cpp:464:31: error: 
      non-constant-expression cannot be narrowed from type 'int' to
      'std::__1::vector<TBTK::Subindex, std::__1::allocator<TBTK::Subindex>
      >::size_type' (aka 'unsigned long') in initializer list
      [-Wc++11-narrowing]
                                vector<Subindex> newFrom({(int)n});
                                                          ^~~~~~
~/TBTK/Lib/src/Builders/ModelFactory.cpp:464:31: note: insert an
      explicit cast to silence this issue
  ...vector<Subindex> newFrom({(int)n});
                               ^~~~~~
                               static_cast<size_type>( )
~/TBTK/Lib/src/Builders/ModelFactory.cpp:464:30: warning: braces
      around scalar initializer [-Wbraced-scalar-init]
                                vector<Subindex> newFrom({(int)n});

```

**Comment by dafer45** [comment]
I do not have a machine where I have been able to reproduce the error. But I have made an update that I hope solves the problem for you.

Do the following in the source folder
```
git pull
git checkout master.v2
```
and then move to the build folder and rerun the installation step
```
cmake ../TBTK
make
sudo make install
```

It is much appreciated if you can let me know whether this fix works.

**Comment by smfarzaneh** [comment]
Thank you @dafer45 for your reply. I tried this fix but there are still errors.  

I don't see the **non-constant-expression** error anymore but other errors like the ones below are showing up while compiling `UnitHandler.h`:  

```shell
[  9%] Building CXX object Lib/src/CMakeFiles/TBTK.dir/Core/Model.cpp.o
In file included from ~/TBTK/Lib/src/Core/Model.cpp:26:
~/TBTK/Lib/include/Utilities/TBTK/UnitHandler.h:435:54: error: constexpr function never produces a constant expression [-Winvalid-constexpr]
inline constexpr Quantity::Angle::Unit& UnitHandler::getUnit<Quantity::Angle>(
```  


```shell
In file included from ~/TBTK/Lib/src/Core/Model.cpp:26:
~/TBTK/Lib/include/Utilities/TBTK/UnitHandler.h:754:39: error: explicit specialization of 'getScale<TBTK::Quantity::Base<TBTK::Quantity::AngleUnit, TBTK::Quantity::AngleExponent> >' after
      instantiation
inline constexpr double& UnitHandler::getScale<Quantity::Angle>(){

```

**Comment by dafer45** [comment]
I have now managed to reproduce the error. It is now fixed and the code has been verified to now compile on Mac. The updated version is available as v2.1.3.

**Comment by smfarzaneh** [comment]
Thanks @dafer45
